app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,22 @@
+class RecipesController < ApplicationController
+require 'will_paginate/array'
+
+  def index
+    recipes = EdamamApiWrapper.list_recipes(params[:query])
+    @recipes = recipes.paginate(:page => params[:page], :per_page => 10)
+  end
+
+  def show
+    if @recipe.nil?
```

Comment:
You haven't tried to find the recipe first, so of course it will be `nil`.

So this show page **will not work**.

Reply:
Correction, this will now work.  Good update